argilla-server/src/argilla_server/validators/records.py

```diff
@@ -71,6 +88,54 @@ def _validate_extra_fields(self, dataset: Dataset, fields: Dict[str, str]) -> No
         if fields_copy:
             raise UnprocessableEntityError(f"found fields values for non configured fields: {list(fields_copy.keys())}")
 
+    def _validate_image_fields(self, dataset: Dataset, fields: Dict[str, str]) -> None:
+        for field in filter(lambda field: field.is_image, dataset.fields):
+            self._validate_image_field(field.name, fields.get(field.name))
+
+    def _validate_image_field(self, field_name: str, field_value: Union[str, None]) -> None:
+        if field_value is None:
+            return
+
+        try:
+            parse_result = urlparse(field_value)
+        except ValueError:
+            raise UnprocessableEntityError(f"image field {field_name!r} has an invalid URL value")
+
+        if parse_result.scheme in ["http", "https"]:
+            return self._validate_web_url(field_name, field_value, parse_result)
+        elif parse_result.scheme in ["data"]:
+            return self._validate_data_url(field_name, field_value, parse_result)
+        else:
+            raise UnprocessableEntityError(f"image field {field_name!r} has an invalid URL value")
+
+    def _validate_web_url(
+        self, field_name: str, field_value: str, parse_result: Union[ParseResult, ParseResultBytes]
+    ) -> None:
+        if not parse_result.netloc or not parse_result.path:
+            raise UnprocessableEntityError(f"image field {field_name!r} has an invalid URL value")
```

Comment:
should we rename "image field" to ImageField in error messaging?

Reply:
We use lower case of class name in all the messages.